ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -223,13 +229,21 @@ public PullQueryResult execute(
           contextStacker,
           pullQueryMetrics);
 
-      return handlePullQuery(
+      final PullQueryResult result = handlePullQuery(
           statement,
           executionContext,
           serviceContext,
           pullQueryContext,
           routingOptions
       );
+
+      if (pullQueryMetrics.isPresent()) {
+        //Record latency at microsecond scale
+        final double latency = (time.nanoseconds() - startTime) / 1000f;
+        pullQueryMetrics.get().recordLatency(latency);
+        pullQueryMetrics.get().recordRate(1);
+      }
```

Comment:
Any reason this can't be pulled into `PullQueryExecutorMetrics`?

```suggestion
      pullQueryMetrics
              .ifPresent(collector -> collector.recordLatency(time.nanoseconds() - startTime));
```

Where `recordLatency` becomes:

```java
public void recordLatency(final long nanoSeconds) {
    //Record latency at microsecond scale
    final double latency = TimeUnit.NANOSECONDS.toMillis(nanoSeconds);
    latencySensor.record(latency);
    requestRateSensor.record(1);
  }
```

And `recordRate` can be deleted.